calling-apis/managing-asynchronous-calls.md

```diff
@@ -0,0 +1,277 @@
+  })
+  .catch( (error) => {
+    console.log('error in findLocation!');
+  });
+}
+
+const seattleCoordinates = findLatitudeAndLongitude('Seattle, Washington, USA');
+
+const locations = findLocation(seattleCoordinates.seattleLat, seattleCoordinates.seattleLon);
+
+console.log(locations);
+```
+<!-- prettier-ignore-end -->
+
+This code:
+
+- Defines `findLatitudeAndLongitude`, which searches for latitude and longitude with the given query
+- Defines `findLocation`, which searches for location with the given latitude and longitude
+- Finds the coordinates of Seattle by calling `findLatitudeAndLongitude` and puts it in `seattleCoordinates`
+- Finds the location results by calling `findLocations` and puts it in `locations`
+
+However, when we run this code, our console outputs this:
+
+```
+error in findLocation!
+success in findLatitudeAndLongitude! 47.6038321 -122.3300624
+```
+
+Raffy's experiment doesn't look successful at all! What's noticeable is that `error in findLocation!` prints out **_before_** `success in findLatitudeAndLongitude!` does.
+
+It's apparent that the API call in `findLatitudeAndLongitude` works. We can get back a latitude and longitude just fine!
+
+Why would there be an error when calling `findLocation`? And why would this print **_before_** we see a print statement from `findLatitudeAndLongitude`?
+
+<!-- Question 1 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: paragraph
+* id: e723527e
+* title: Managing Asynchronous Calls
+##### !question
+
+Make a hypothesis:
+
+Why would there be an error when calling `findLocation`? And why would this print **_before_** we see a print statement from `findLatitudeAndLongitude`?
+
+##### !end-question
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+### Debugging Our Problem
+
+After debugging, Raffy should hone in on one specific line:
+
+```js
+const locations = findLocation(seattleCoordinates.seattleLat, seattleCoordinates.seattleLon);
+```
+
+Raffy found these two facts while debugging:
+
+- `seattleCoordinates.seattleLat` _and_ `seattleCoordinates.seattleLon` are `undefined` at the time of calling `findLocation`
+- The print statements in `findLocation` printed, meaning that `findLocation` was definitely called and got into the `catch` block
+
+After diagramming, thinking, researching, and rubber-ducking, Raffy comes to this conclusion:
+
+- `seattleCoordinates` has undefined values because it hasn't been assigned a value yet, and JavaScript gives `undefined` when accessing `seattleCoordinates.seattleLat` and `seattleCoordinates.seattleLon`
+- `seattleCoordinates` is assigned a value from `findLatitudeAndLongitude`'s return
+- `seattleCoordinates` wasn't assigned a value because `findLatitudeAndLongitude` didn't return yet
+- Something that changes the flow of control in our code is our asynchronous API calls
+
+### Refactoring: Utilize `then`
+
+How can Raffy ensure that `findLocation` executes after `findLatitudeAndLongitude` finishes?
```

Comment:
incorporate latitude and longitude into this discussion, since they're currently the source of the undefined values
```suggestion
How can Raffy ensure that `findLocation`, which needs the `latitude` and `longitude`, executes only after the API call in `findLatitudeAndLongitude` finishes?
```